woocommerce/wc-admin
# Global Header & Activity Panel on all WooCommerce Pages

This branch makes the header and activity panel embeddable on existing core WooCommerce pages.

Closes #96.
See p6riRB-3en-p2 #comment-3907.

Two major changes:

WordPress notifications are also rolled up into the activity panel like they are on new React based pages. The code is updated to pull out success / error messages and to correctly deal with the screen options tabs. 

Breadcrumb handling has been approved. Some of the functions here for setting core breadcrumbs, I am considering temporary. When we merge this project to WooCommerce, we should be able to provide some registration functions and actually call them from core code.

This is also not enabled for extensions by default, but they can opt in.

Finally, the new ActivityPanel design is not implemented here. I've added a few todos around this -- but for now, the breadcrumbs and notices behavior is more important to test.

To Test:
* Run `npm start`. There is a webpack change here, so make sure to restart it after pulling down the branch.
* Visit the current woo-dash page and make sure everything still looks OK there.
* Visit any WooCommerce page and see the header embedded.
* Click around to different pages and check breadcrumbs.
* Test the notices toggle.
* Test screen help buttons expanding off the embedded header.
* Try creating or updating an order and test that the success message or JITM message are not rolled up behind the WP notices toggle.

<img width="1264" alt="screen shot 2018-06-25 at 11 16 51 am" src="https://user-images.githubusercontent.com/689165/41859055-4c5c1b20-7869-11e8-8aac-b7cf28543a17.png">


---

Previous PR Description:

> If we move forward with this, there are a number of things we’d probably want to clean up (for example I had to move some CSS around to get stuff working) and refactor based on this branch.

> Here are a few issues/notes based on the approach:

> * Success messages (like “order created”) are rolled up behind the WordPress notice toggle. How would we handle this on existing/classic WooCommerce pages? 
> * Breadcrumbs are naive and currently using the page/menu title. So all settings pages get “WooCommerce / Settings”. If we were serious about embedding this, we should develop a system for registering better section names. We could use the current menu/page name as a fallback.
> * Position of the “Screen Options” and “Help” tabs. I used some CSS to move these down and make them look better since they are now separated from the admin-bar. We would need to consider placement of these for classic pages.
> * Uncertainty of what other extensions are doing / how it will interact. If we move forward with something like this, maybe we could enable this for core and our extensions we’ve tested with and then make it opt-in for other extensions? This would give authors a chance to make their interfaces play nice with the new header.

> Also note that the notice toggle doesn’t correctly count Jetpack JITM messages and some “woocommerce-message” notices without the notice class. I’m logging a separate issue for this.
